### PR TITLE
chore: Update dependencies

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
`eslint@6.x` [requires](https://www.npmjs.com/package/eslint/v/6.0.0#installation-and-usage) `node@^8.10`, so I updated the minimum required version in `engines` and tests

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,39 +26,46 @@
...
+    "extends": "./index.js"
+  },
+  "eslintIgnore": [
+    "node_modules",
+    "coverage",
+    "dist"
+  ],
   "engines": {
-    "node": ">=8",
+    "node": ">=8.10",
```
Comment (kentcdodds) [inline package.json:68]: Why this change? Maybe we should just drop Node 8 instead?

File: package.json [hunk 2]
```diff
@@ -4,24 +4,19 @@
...
+    "find-new-rules:main": "eslint-find-rules --unused ./index.js",
     "find-new-rules:react": "eslint-find-rules --unused ./other/test/react/index.js",
     "find-new-rules:webpack": "eslint-find-rules --unused ./other/test/webpack/index.js",
-    "validate": "npm t"
-  },
-  "husky": {
-    "hooks": {
-      "pre-commit": "pretty-quick --staged && npm run validate"
-    }
+    "lint": "kcd-scripts lint",
```
Comment (kentcdodds) [inline package.json:13]: Oh, I want to use this project's lint config to lint itself as an additional test.

File: package.json (hunk 1 above)
Comment (MichaelDeBoey) [reply package.json:68]: > Why this change?

`eslint@6.x` [requires](https://www.npmjs.com/package/eslint/v/6.0.0#installation-and-usage) `node@^8.10`


> Maybe we should just drop Node 8 instead?

I think it's probably a good idea to wait for ESLint 7, since that will drop Node 8.

https://github.com/eslint/eslint/releases/tag/v7.0.0-alpha.0

File: package.json (hunk 2 above)
Comment (MichaelDeBoey) [reply package.json:13]: I've set the linting config to `./index.js`, so this project will lint itself

Comment (kentcdodds) [reply package.json:13]: Oh. Missed that.
